rustls/src/msgs/message.rs

```diff
@@ -28,10 +28,12 @@ impl MessagePayload {
         }
     }
 
-    pub fn new(typ: ContentType, vers: ProtocolVersion, payload: Payload) -> Result<Self, Error> {
-        let mut r = Reader::init(&payload.0);
+    pub fn new(typ: ContentType, vers: ProtocolVersion, payload: Vec<u8>) -> Result<Self, Error> {
```

Comment:
Is there a way to make some subset of the progress here without baking in `Vec<u8>` where `Payload` was before? It seems like a step backwards to have `Vec` in more places now, given other conversations.

Reply:
I still think this is progress along that path: it simply removes one layer of indirection. I did have a big PR that ends up changing Vec to Cow but rebasing that was a pain so I'd prefer to land smaller pieces first.

Reply:
This part of the PR was removed.